src/apps/common/submodules/portWizard/portWizard.js

```diff
@@ -1009,9 +1062,18 @@ define(function(require) {
 		 */
 		portWizardOwnershipConfirmationRender: function(args, callback) {
 			var self = this,
+				requiredDocumentsCompleteList = self.portWizardGet('requiredDocumentsList'),
+				invoiceDocumentMetadata = _.find(requiredDocumentsCompleteList, { key: 'Invoice' }),
+				buildTemplateArgs = _
+					.chain(args)
+					.pick('data')
+					.merge({
+						invoiceDocumentMetadata: invoiceDocumentMetadata
+					})
+					.value(),
```

Comment:
On issue with this chaining is the fact that `buildTemplateArgs.data` holds a reference of `args.data` (shallow vs deep copy).

What you are doing:
```js
var buildTemplateArgs = {
    data: args.data,
    invoiceDocumentMetadata: invoiceDocumentMetadata
};
```

What you want to be doing:
```js
var buildTemplateArgs = {
    data: { ..args.data }
    invoiceDocumentMetadata: invoiceDocumentMetadata
};
```
I am using object spread here for conciseness but in lodash world this could be done by using your picked object as source rather than destination of your merge:
```js
var buildTemplateArgs = _.merge({
    invoiceDocumentMetadata: invoiceDocumentMetadata
}, _.pick(args, 'data'));
```

Reply:
In this block I wanted to have a shalllow copy as a way to save some memory space, because the `args.data` object is only read and never modified in the `GetTemplate` functions below. Is it OK, or a deep copy should always be created in this kind of scenarios, to prevent any unintended side effects?

Reply:
> Is it OK, or a deep copy should always be created in this kind of scenarios, to prevent any unintended side effects?

Keeping your data immutable (and, as you noted, prevent side effects) trumps memory complexity in almost any scenario; edge cases usually involve recursion, in which case specific optimization techniques can be implemented when your algorithm becomes a memory hog (memoization or switching to a loop statement).

> In this block I wanted to have a shalllow copy as a way to save some memory space, because the `args.data` object is only read and never modified in the `GetTemplate` functions below.

While this idea makes sense now, it is because this part of the codebase is fresh in your memory. Imagine having to come back to it in three months, or someone else modifying the structure of `buildTemplateArgs` downstream without realizing `args.data` will be modified as well.